File: dbflow-rx2/src/main/java/com/raizlabs/android/dbflow/rx2/language/RXSQLite.java
package com.raizlabs.android.dbflow.rx2.language;

import android.support.annotation.NonNull;

import com.raizlabs.android.dbflow.sql.language.SQLite;
import com.raizlabs.android.dbflow.sql.queriable.ModelQueriable;
import com.raizlabs.android.dbflow.sql.queriable.Queriable;

/**
 * Description: The RX implementation of SQLite language queries. Meant to be interchangeable
 * with {@link SQLite}.
 */
public class RXSQLite {

    @NonNull
    public static <T> RXModelQueriableImpl<T> rx(ModelQueriable<T> modelQueriable) {
        return new RXModelQueriableImpl<>(modelQueriable);
    }

    @NonNull
    public static RXQueriableImpl rx(Class<?> table, Queriable queriable) {
        return new RXQueriableImpl(table, queriable);
    }

}

File: dbflow-rx2/src/main/java/com/raizlabs/android/dbflow/rx2/language/RXSQLite.kt
@file:JvmName("RXSQLite")

package com.raizlabs.android.dbflow.rx2.language

import com.raizlabs.android.dbflow.query.ModelQueriable
import com.raizlabs.android.dbflow.query.Queriable

fun <T : Any> ModelQueriable<T>.rx(): RXModelQueriableImpl<T> = RXModelQueriableImpl(this)

fun Queriable.rx(): RXQueriableImpl = RXQueriableImpl(this)

